pkg/clustercache/clustercache.go

```diff
@@ -118,11 +133,52 @@ type PersistentVolume struct {
 	Status      v1.PersistentVolumeStatus
 }
 
-type ReplicationController struct{}
+type ReplicationController struct {
+	Name      string
+	Namespace string
+	Spec      v1.ReplicationControllerSpec
+}
+
+type PodDisruptionBudget struct {
+	Name      string
+	Namespace string
+	Spec      policyv1.PodDisruptionBudgetSpec
+	Status    policyv1.PodDisruptionBudgetStatus
+}
 
-type PodDisruptionBudget struct{}
 type ReplicaSet struct {
+	Name         string
+	Namespace    string
 	SpecSelector *metav1.LabelSelector
+	Spec         appsv1.ReplicaSetSpec
+}
+
+type Volume struct {
+}
+
+// GetControllerOf returns a pointer to a copy of the controllerRef if controllee has a controller
+func GetControllerOf(pod *Pod) *metav1.OwnerReference {
```

Comment:
I don't really like copying things around unless it's necessary. It's also can be difficult to read. Maybe something like generic `deepCopy` do the job?

I think copying is the main reasons of high memory usage on querying opencost.

Reply:
This function is actually straight from the k8s API, but leveraging the opencost types. See here: https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/controller_ref.go